src/gvmd.c

```diff
@@ -3411,13 +3436,19 @@ gvmd (int argc, char** argv, char *env[])
       gvm_close_sentry ();
       exit (EXIT_FAILURE);
     }
-
+#if OPENVASD
+  if (openvasd_scanner_exists ())
+    {
+      g_critical ("%s: No openvasd VT update client found.",
+                  __func__);
+#else
   if (check_osp_vt_update_socket ())
     {
       g_critical ("%s: No OSP VT update socket found."
                   " Use --osp-vt-update or change the 'OpenVAS Default'"
                   " scanner to use the main ospd-openvas socket.",
                   __func__);
+#endif
```

Comment:
I suggest to just skip the socket check if openvasd is enabled
```suggestion
#if OPENVASD == 0
  if (check_osp_vt_update_socket ())
    {
      g_critical ("%s: No OSP VT update socket found."
                  " Use --osp-vt-update or change the 'OpenVAS Default'"
                  " scanner to use the main ospd-openvas socket.",
                  __func__);
#endif
```